rad/prelude/patterns.rad

```diff
@@ -227,3 +228,29 @@
           [1 (/member [1 3 5])] :no
           [1 (/member [2 4 8])] :yes
           _                     :no) ==> :yes])
+
+(def /number
+  "Matches values which are numbers. Matches are passed to the inner pattern."
+  (fn [pat]
+    (fn [v]
+      (if (number? v)
+        (match-pat pat v)
+        :nothing))))
+
+(def /integer
+  "Matches numbers which are integral. Matches are passed to the inner pattern."
+  (fn [pat]
+    (fn [v]
+      (if (integral? v)
+        (match-pat pat v)
+        :nothing))))
+
+(def /number-string
+  "Matches strings which can be parsed as numbers. The parsed number is passed to
+  the inner pattern."
+  (fn [pat]
+    (fn [s]
+      (def n (read s))
```

Comment:
Maybe a `catch` here, since otherwise we get a nothing when it parses as something other than number, but an exception if it doesn't parse as anything at all, which seems confusing.